### PR TITLE
WFCORE-4868 Introduce aliases for standard configuration files

### PR DESCRIPTION
**Pull request by michpetrov**
Issue: [EAP7-1700](https://issues.redhat.com/browse/EAP7-1700) / [WFCORE-4868](https://issues.redhat.com/browse/WFCORE-4868)

### CODE REVIEW

File: controller/WFCORE-4868_configuration-files-aliases.adoc [hunk 1]
```diff
@@ -0,0 +1,98 @@
+---
+categories:
+ - controller
+---
+= Aliases for standard configuration files
```
Comment (yersan) [inline controller/WFCORE-4868_configuration-files-aliases.adoc:5]: @michpetrov We need to add the  [Community] word here in the feature title, that will help when this proposal is published on wildfly.org

File: controller/WFCORE-4868_configuration-files-aliases.adoc [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+| *-microprofile-ha.xml  | mpha
+|===
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE-4868
+* https://issues.redhat.com/browse/EAP7-1700
```
Comment (yersan) [inline controller/WFCORE-4868_configuration-files-aliases.adoc:39]: The EAP7 reference should be removed from this doc, since this is community RFE

**Comment by yersan** [comment]
> It is not clear to me where this aliases list live.
So far it is only available for the configuration files we provide, so it is hardcoded in the code, maybe they could be extracted in an automatic and predictable way, but that is not available for the first cut yet.

> i would love to make it editable so I could add my own aliases :) Also does it only concerns sandlone configuration files ? 
That could be an evolution of this one. Right now is only prepared to supply aliases of the configuration files we provide by default. Those are the standalone variants.

File: controller/WFCORE-4868_configuration-files-aliases.adoc [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+| *-full-ha.xml          | fha
+
+| *-load-balancer.xml    | lb
+
+| *-microprofile.xml     | mp
+
+| *-microprofile-ha.xml  | mpha
+|===
+
+Special values for `-c` remain in effect, e.g. `-c boot` references the standalone.boot.xml file (if present) and it is not possible to reference a custom file named `standalone-boot.xml` this way.
```
Comment (yersan) [inline controller/WFCORE-4868_configuration-files-aliases.adoc:35]: I am more inclined to keep the `boot` (and `initial`, v* and so on) untouched as they are. They should not be treated as possible aliases. If I use `-c boot`, this `boot` should not be treated as an alias, it should have the same meaning as before.

File: controller/WFCORE-4868_configuration-files-aliases.adoc [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+* [ ] Preview
+
+* [x] Community
+
+* [ ] default
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
```
Comment (yersan) [inline controller/WFCORE-4868_configuration-files-aliases.adoc:55]: @michpetrov This section describes the feature team. 

Please add you as a developer, me as SME, and Emmanuel Hugonnet as an outside perspective person.

File: controller/WFCORE-4868_configuration-files-aliases.adoc (hunk 3 above)
Comment (michpetrov) [reply controller/WFCORE-4868_configuration-files-aliases.adoc:35]: That's why I'm specifying that `boot` _cannot_ be used as an alias for a file named "standalone-boot.xml"

Comment (yersan) [reply controller/WFCORE-4868_configuration-files-aliases.adoc:35]: Ahh, got it @michpetrov , I wrongly read this statement, thanks!

**Comment by yersan** [comment]
> Updated.
> 
> Just to clarify and perhaps I need to make the distinction clearer. I want this to work for the file suffixes, and a set of strings that are meant for the files we distribute.
> 
> I.e. "standalone-load-balancer.xml" can be referenced by "lb" (which will be hardcoded, and we could make this extensible in the future) but it could also be referenced by "load-balancer" (which is not hardcoded).

@michpetrov You are proposing to also enable other aliases besides the ones that can be applied to the files we distribute. So if the user has `standalone-custom.xml` they can reference it using just `-c custom`. That's more versatile, but I don't see it too useful at all, instead of creating `standalone-custom.xml` you could create `custom.xml`, even more, if you plan to restrict all to [`standalone` as the base name](https://github.com/wildfly/wildfly-core/pull/5903#discussion_r1529138040). Creating the shorter file seems simpler and avoids any weird behavior when you have `standalone-custom.xml` and `custom.xml` both on your configuration file. In such a case, if you use `-c custom` it is a bit confusing the expected logic about which file will be finally picked up. I would rather go with the simplest option which would be to translate with aliases the files we distribute only. In the future, if the need arises, we can think of how to allow the users to create custom aliases. I have no strong opinion on that, though, but I am more inclined to treat only the files we distribute.

If we go as it is not, could you also clarify this case in the proposal and define what would be the final logic applied?

**Comment by darranl** [comment]
FYI if we are targeting community the "Nice-to-have" requirements should be converted to non-requirements or moved to a future-work section

**Review by yersan — APPROVED** [review]

**Review by ehsavoie — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @michpetrov, @yersan and @ehsavoie!

**Pull request merged**